gradio/demo.py

```diff
@@ -0,0 +1,274 @@
+    if semantic_id in dataset_info.thing_list:
+      # For `thing` class, we will add a small amount of random noise to its
+      # correspondingly predefined semantic segmentation colormap.
+      unique_instance_ids = np.unique(instance_map[semantic_mask])
+      for instance_id in unique_instance_ids:
+        instance_mask = np.logical_and(semantic_mask,
+                                       instance_map == instance_id)
+        random_color = perturb_color(
+            dataset_info.colormap[semantic_id],
+            perturb_noise,
+            used_colors[semantic_id],
+            random_state=random_state)
+        colored_panoptic_map[instance_mask] = random_color
+    else:
+      # For `stuff` class, we use the defined semantic color.
+      colored_panoptic_map[semantic_mask] = dataset_info.colormap[semantic_id]
+      used_colors[semantic_id].add(tuple(dataset_info.colormap[semantic_id]))
+  return colored_panoptic_map, used_colors
+
+
+def vis_segmentation(image,
+                     panoptic_prediction,
+                     dataset_info,
+                     perturb_noise=60):
+  """Visualizes input image, segmentation map and overlay view."""
+  plt.figure(figsize=(30, 20))
+  grid_spec = gridspec.GridSpec(2, 2)
+
+  ax = plt.subplot(grid_spec[0])
+  plt.imshow(image)
+  plt.axis('off')
+  ax.set_title('input image', fontsize=20)
+
+  ax = plt.subplot(grid_spec[1])
+  panoptic_map, used_colors = color_panoptic_map(panoptic_prediction,
+                                                 dataset_info, perturb_noise)
+  plt.imshow(panoptic_map)
+  plt.axis('off')
+  ax.set_title('panoptic map', fontsize=20)
+
+  ax = plt.subplot(grid_spec[2])
+  plt.imshow(image)
+  plt.imshow(panoptic_map, alpha=0.7)
+  plt.axis('off')
+  ax.set_title('panoptic overlay', fontsize=20)
+
+  ax = plt.subplot(grid_spec[3])
+  max_num_instances = max(len(color) for color in used_colors.values())
+  # RGBA image as legend.
+  legend = np.zeros((len(used_colors), max_num_instances, 4), dtype=np.uint8)
+  class_names = []
+  for i, semantic_id in enumerate(sorted(used_colors)):
+    legend[i, :len(used_colors[semantic_id]), :3] = np.array(
+        list(used_colors[semantic_id]))
+    legend[i, :len(used_colors[semantic_id]), 3] = 255
+    if semantic_id < dataset_info.num_classes:
+      class_names.append(dataset_info.class_names[semantic_id])
+    else:
+      class_names.append('ignore')
+
+  plt.imshow(legend, interpolation='nearest')
+  ax.yaxis.tick_left()
+  plt.yticks(range(len(legend)), class_names, fontsize=15)
+  plt.xticks([], [])
+  ax.tick_params(width=0.0, grid_linewidth=0.0)
+  plt.grid('off')
+#   plt.show()
+  return plt
+
+def run_cmd(command):
+    try:
+        print(command)
+        call(command, shell=True)
+    except KeyboardInterrupt:
+        print("Process interrupted")
+        sys.exit(1)
+MODEL_NAME = 'max_deeplab_l_backbone_os16_axial_deeplab_cityscapes_trainfine_saved_model'   
```

Comment:
would it be possible for users to select which model to use?

btw the default model here (maxdeeplab l) is fairly heavy. Would a lighterweight model be better as default? (e.g resnet50_os32_panoptic_deeplab_cityscapes_crowd_trainfine_saved_model)

Reply:
Hi sure i can try that currently using cpu on gradio hub so that is why inference is slow, I will update with this one to see if that helps

Reply:
that is alot faster thanks!
<img width="1020" alt="Screen Shot 2021-06-28 at 7 34 02 PM" src="https://user-images.githubusercontent.com/81195143/123716126-d43d5900-d847-11eb-9db9-d7008df942ac.png">

Reply:
in terms of the user choosing the model, I think lightweight models would be better for user experience, so I can create a dropdown that only include the lightweight ones

Reply:
yes a dropdown would be great. optionally we can also include larger ones and note users that could take a long time.